plugins/outputs/opentelemetry/client.go

```diff
@@ -0,0 +1,217 @@
+		// For coralogix, we enforce HTTP connection with TLS
+		h.httpClient.Transport = &http.Transport{
+			TLSClientConfig: &ntls.Config{},
+		}
+	} else {
+		h.httpClient.Transport = &http.Transport{
+			TLSClientConfig: &ntls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+	return nil
+}
+
+func (g *gRPCClient) Export(ctx context.Context, request pmetricotlp.ExportRequest) (pmetricotlp.ExportResponse, error) {
+	return g.metricsServiceClient.Export(ctx, request, g.callOptions...)
+}
+
+func (g *gRPCClient) Close() error {
+	if g == nil {
+		return nil
+	}
+
+	if g.grpcClientConn != nil {
+		err := g.grpcClientConn.Close()
+		g.grpcClientConn = nil
+		return err
+	}
+	return nil
+}
+
+func (h *httpClient) Export(ctx context.Context, request pmetricotlp.ExportRequest) (pmetricotlp.ExportResponse, error) {
+	var err error
+	var requestBytes []byte
+
+	switch h.encodingType {
+	case "application/x-protobuf":
+		requestBytes, err = request.MarshalProto()
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	case "application/json":
+		requestBytes, err = request.MarshalJSON()
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	default:
+		return pmetricotlp.ExportResponse{}, fmt.Errorf("unsupported content type '%s'", h.encodingType)
+	}
+	var reader io.Reader
+	reader = bytes.NewReader(requestBytes)
+
+	if h.compress != "" && h.compress != "none" {
+		reader = internal.CompressWithGzip(reader)
+	}
+
+	httpRequest, err := http.NewRequestWithContext(ctx, http.MethodPost, h.url, reader)
+	if err != nil {
+		return pmetricotlp.ExportResponse{}, err
+	}
+
+	httpRequest.Header.Set("Content-Type", h.encodingType)
+	httpRequest.Header.Set("User-Agent", userAgent)
+	if h.compress != "" && h.compress != "none" {
+		httpRequest.Header.Set("Content-Encoding", "gzip")
+	}
+
+	httpResponse, err := h.httpClient.Do(httpRequest)
+	if err != nil {
+		return pmetricotlp.ExportResponse{}, err
+	}
+	defer func() {
+		//nolint:errcheck // cannot fail with io.Discard
+		io.CopyN(io.Discard, httpResponse.Body, maxHTTPResponseReadBytes)
+		_ = httpResponse.Body.Close()
+	}()
+
+	if httpResponse.StatusCode < 200 || httpResponse.StatusCode >= 300 {
+		return pmetricotlp.ExportResponse{}, fmt.Errorf("received non-2xx HTTP response code: %d", httpResponse.StatusCode)
+	}
+
+	responseBytes, err := readResponseBody(httpResponse)
+	if err != nil {
+		return pmetricotlp.ExportResponse{}, err
+	}
+
+	exportResponse := pmetricotlp.NewExportResponse()
+	switch httpResponse.Header.Get("Content-Type") {
+	case "application/x-protobuf":
+		err = exportResponse.UnmarshalProto(responseBytes)
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	case "application/json":
+		err = exportResponse.UnmarshalJSON(responseBytes)
+		if err != nil {
+			return pmetricotlp.ExportResponse{}, err
+		}
+	}
+
+	return exportResponse, nil
+}
+
+func (*httpClient) Close() error {
+	// No persistent connections to close for HTTP client
```

Comment:
I think you need to [`CloseIdleConnections`](https://pkg.go.dev/net/http#Client.CloseIdleConnections) here.

Reply:
Fixed. https://github.com/influxdata/telegraf/pull/17997/commits/4ecf9758badaeae5324168ff38d85d1b5e48a723